### DISCUSSION TITLE
Credentials - PHP 8.X

### DISCUSSION THREAD

**Discussion by saruman** [aws/aws-sdk-php#2724] [Q&A] [answered]
Hi.
I have been using the SDK for months with no issues to send emails (SES), but after I updated the SDK to the latest version i got an error.  Can anyone help me to figure this out?

InvalidArgumentException [ 0 ]: Invalid configuration value provided for "configuration_mode". Expected Aws\DefaultsMode\ConfigurationInterface|Aws\CacheInterface|string|closure, but got array(2) configuration_mode: (Aws\DefaultsMode\ConfigurationInterface|Aws\CacheInterface|string|closure) Sets the default configuration mode. Otherwise provide an instance of Aws\DefaultsMode\ConfigurationInterface, an instance of Aws\CacheInterface, or a string containing a valid mode

```
<?php

use Aws\Ses\SesClient;

$client = new SesClient([
    'version' => 'latest',
    'region' => 'REGION',
    'credentials' => [
        'accessKeyId' => 'YOUR_ACCESS_KEY',
        'secretAccessKey' => 'YOUR_SECRET_ACCESS_KEY'
    ]
]);
```

**Comment by SamRemis** [comment]
Hi there,
This should maybe be filed as a github issue, but I wasn't able to reproduce your error.  When I used the same inputs as above on version 8.2 of PHP, I got a different error: 

`Credentials must be an instance of 'Aws\Credentials\CredentialsInterface, an associative array that contains "key", "secret", and an optional "token" key-value pairs, a credentials provider function, or false.`

That makes sense, the keys to your credentials array should be `key` and `secret`, not `accessKeyId` and `secretAccessKey`.  I'm also going to go on the assumption that you replaced `'REGION'` above with a valid region like `us-east-2`, otherwise you'd be getting CURL errors.  

I'm wondering if it's possible that you somehow have accidentally set the `configuration_mode` input parameter somewhere; either in a part of the client constructor that isn't shown, in another client constructor elsewhere in the code, or if you've set the `AWS_DEFAULTS_MODE` environment variable or the ini key `defaults_mode`.  

To confirm, you're using version 3.275.5 of the SDK?

**Reply by saruman** [reply]
Thank you @SamRemis for reply.
This is exactly the code that Im using.  My bad for the previous example, I took it from someone else here.
I don't quite understand the part that you mentioned about somehow accidentally set the configuration_mode.  Where exactly is that? I searched for that param and didnt found it.  Besides that, I don't know what could be my issue.
And yes, Im using latest version 3.275.5 with PHP 8.1
```
<?php

use Aws\Ses\SesClient;

$client = new SesClient([
    'version' => 'latest',
    'region' => 'us-east-1',
    'credentials' => [
        'key' => 'YOUR_ACCESS_KEY',
        'secret' => 'YOUR_SECRET_ACCESS_KEY'
    ]
]);
```

**Comment by SamRemis** [comment]
Then it sounds like there's either a glitch of some sort or something going on in your environment.  Can you confirm that you're using version 3.275.5 and tell me what type of system you're running the code on?

**Reply by saruman** [reply]
I updated my last comment, forgot to put the version.
And yes, Im using latest version 3.275.5 with PHP 8.1.  Im going to start from scratch using a fresh composer script, because what I did was replace the old sdk for the new one from github.  I'll let you know in a momment.

**Reply by saruman** [reply, marked as answer]
@SamRemis the problem was that I didn't rebuild the sdk using composer.  Now is working.
Thank you.

[a comment by an automated account is not shown]
